src/qml/RelationCombobox.qml

```diff
@@ -134,59 +135,19 @@ Item {
     }
   }
 
-  AttributeFormModel {
-   id: attributeFormModel
-   featureModel: FeatureModel {
-       currentLayer: relationCombobox._relation ? relationCombobox._relation.referencedLayer : null
-     }
-  }
-
-  Loader {
-    id: addFeatureForm
-    sourceComponent: addFeatureFormComponent
-    active: false
-    onLoaded: {
-      item.open()
-    }
-  }
-
-  Component {
-    id: addFeatureFormComponent
-    Popup {
-      id: popup
-      parent: ApplicationWindow.overlay
-
-      x: 24 * dp
-      y: 24 * dp
-      width: parent.width - 48 * dp
-      height: parent.height - 48 * dp
-      padding: 0
-      modal: true
-      focus: true
-      closePolicy: Popup.CloseOnEscape
-
-      FeatureForm {
-        model: attributeFormModel
-
-        anchors.fill: parent
-
-        state: "Add"
-        embedded: true
-
-        onSaved: {
-          var referencedValue = attributeFormModel.attribute(relationCombobox._relation.resolveReferencedField(field.name))
-          comboBox.currentIndex = featureListModel.findKey(referencedValue)
-          popup.close()
-        }
+  EmbeddedFeatureForm{
+      id: addFeaturePopup
 
-        onCancelled: {
-          popup.close()
-        }
+      onFeatureCancelled: {
+          if( autoSave ) {
+              var referencedValue = addFeaturePopup.attributeFormModel.attribute(relationCombobox._relation.resolveReferencedField(field.name))
+              comboBox.currentIndex = featureListModel.findKey(referencedValue)
+          }
       }
 
-      onClosed: {
-        addFeatureForm.active = false
+      onFeatureSaved: {
+          var referencedValue = addFeaturePopup.attributeFormModel.attribute(relationCombobox._relation.resolveReferencedField(field.name))
+          comboBox.currentIndex = featureListModel.findKey(referencedValue)
```

Comment:
Here we have an issue on *not* auto save mode. This is happening before the featureListModel is reloaded (with timer) what means that it cannot find the value...